### PR TITLE
FEA-3682: v4

### PR DESCRIPTION
**Pull request by matthewnitschke-wk**
# [FEA-3682](https://jira.atl.workiva.net/browse/FEA-3682)
Issue Status

## Motivation

Includes changes from both:
- #107
- #108

Any current consumers of dependency_validator might encounter failures due to these changes, merging into a single v4 branch, and will be released as a major to specifically inform the "breaking" changes

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: .github/workflows/dart_ci.yaml [hunk 1]
```diff
@@ -12,21 +12,17 @@ on:
...
+          sdk: 2.19.6
       - name: Install dependencies
         run: dart pub get
       - name: Validate dependencies
         run: dart run dependency_validator
-      - name: Check formatting
-        run: dart format --output=none --set-exit-if-changed .
-        if: ${{ matrix.sdk == 'stable' }}
+      # - name: Check formatting
+      #   run: dart format --output=none --set-exit-if-changed .
```
Comment (todbachman-wf) [inline .github/workflows/dart_ci.yaml:25]: Is there a reason why you've commented out this check?

File: test/executable_test.dart [hunk 2]
```diff
@@ -411,6 +411,14 @@ void main() {
         expect(result.stderr, contains('fake_project'));
       });
 
+      test('and import is commented out', () async {
+         await d.dir('unused', [
+           d.dir('lib', [
+            d.file('invalid.dart', '// import \'package:fake_project/fake.dart\';'), // commented out import
+          ]),
+        ]).create();
```
Comment (todbachman-wf) [inline test/executable_test.dart:419]: It seems like this test doesn't have any checks in it.

File: .github/workflows/dart_ci.yaml (hunk 1 above)
Comment (matthewnitschke-wk) [reply .github/workflows/dart_ci.yaml:25]: Ah, sorry, that got a bit lost in the v4 branch: https://github.com/Workiva/dependency_validator/pull/107#discussion_r1520192455

**Review by todbachman-wf — DISMISSED** [review]

File: test/executable_test.dart [hunk 3]
```diff
@@ -647,7 +696,7 @@ void main() {
             version: 0.0.0
             private: true
             environment:
-              sdk: '>=2.4.0 <3.0.0'
+              sdk: '>=2.12.0 <4.0.0'
```
Comment (matthewnitschke-wk) [inline test/executable_test.dart:699]: bounds of these were expanded so we can run pub get with dart 3

**Review by todbachman-wf — APPROVED** [review]

**Review by evanweible-wf — APPROVED** [review]

**Comment by evanweible-wf** [comment]
semver +1

File: .github/workflows/dart_ci.yaml [hunk 4]
```diff
@@ -12,21 +12,17 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        sdk: [ 2.19.6 ]
```
[the remark this answers is not in the document]
Comment (matthewnitschke-wk) [reply .github/workflows/dart_ci.yaml]: Looks like this comment is a bit old, we're running on stable now: https://github.com/Workiva/dependency_validator/pull/111/files#diff-0530c3bf28665689d37061f6d19aade372700afec3ae45decc3f2edbd64e344dR18

**Comment by matthewnitschke-wk** [comment]
QA +1

- [x] CI does indeed pass
- [x] Ran some quick smoke tests in a few repos to verify existing functionality

🚀 @Workiva/release-management-p 🚢

**Review by rmconsole-wf — APPROVED** [review]
+1 from RM

**Pull request merged**